File: Programmers/Lv_0/marble.js
function fac(n) {
    let bigN = BigInt(n);

    if (bigN === BigInt(1) || bigN === BigInt(0)) {
        return BigInt(1);
    }

    return bigN * fac(bigN - BigInt(1));
}

function solution(balls, share) {
    const n = balls;
    const m = share;

    return fac(n) / (fac(n-m) * fac(m));
}

File: Programmers/Lv_0/marble.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int fac(int n) {
    unsigned long long answer = 1;
    for (int i = 2; i <= n; i++) {
        answer *= i;
    }
    return answer;
}

int solution(int balls, int share) {
    unsigned long long answer = 0;

    answer = fac(balls) / (fac(balls-share) * fac(share));

    return answer;
}

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int balls, share;
    cin >> balls >> share;
    cout << solution(balls, share);

    return 0;
}